### PR TITLE
Staff Index and Create Page

### LINKED ISSUE #2: EPIC: Staff CRUD operations (multiple stories, some of which can be done in parallel)
# User Story

* As an admin
* I can perform Create, Read, Update, Delete operations on staff
* so that I can manage the staff's for that particular course without having to go through swagger

### Discussion

The Staff Backend API for CRUD operations is  already working and available on the swagger and controllers. 

However, we don't yet have a frontend.  We need a frontend, similar, probably to the one you build in team03 

### Acceptance Criteria

NOTE: it is strongly recommended to break this into multiple issues: for example, as team03 was broken into issues, and do separate PRs for each part.  You could start by just creating a form and a table, and different team members could be assigned to do each of those.

As Admin:
- [ ] I can go to the staff page from the courses page.
- [ ] I can list all staff members
- [ ] I can add a staff member
- [ ] I can update a staff members information
- [ ] I can delete a staff member

As User:
- [ ] I can list all of the staff for a course.
- [ ] I cannot add, edit or update staff for a course.

### PR DESCRIPTION
**Pull request by Peak1260**
In this pull request, I created the interfaces for the staff index page. Once an admin or instructor clicks on the staff button on the courses page, it will link to this one. The tests files are all complete and I created an add staff button, along with a delete button for further implementation of the staff page. Authorized users will now have access to adding staff by the courseId and the GitHub Login field. 

https://proj1-peak1260-dev1.dokku-09.cs.ucsb.edu/

### CODE REVIEW

**Comment by Peak1260** [comment]
I don't quite know why but upon further testing, it seems like when I click the delete button it does delete the staff, but the row is still there. However, if you refresh the page the row goes away. If someone can help me on this issue, it will be greatly appreciated. (You can see the delete button does indeed work as well because when you click again it shows up an error saying the id is already deleted).

Update: Upon asking the TAs in the section, we still haven't figured it out. I have to refresh the page every time for it to reflect. I wish it would cleanly delete the same way courses page does, but I'm completely stuck as to what else I can do.

**Review by ylin250 — DISMISSED** [review]
LGTM!

File: frontend/src/main/pages/StaffIndexPage.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+export default function StaffIndexPage() {
+  let {id} = useParams();
+
+  const { data: currentUser } = useCurrentUser();
+  const createButton = () => {  
+
+      return (
+          <Button
+              variant="primary"
+              href="/staff/create"
```
Comment (pconrad) [inline frontend/src/main/pages/StaffIndexPage.js]: This is not the right url.  There is no notion of creating a staff member outside the context of a specific course.  So the URL should be:

```suggestion
              href=`/staff/${id}/create`
```

And you need at least a temporary placeholder page that says: 
`Placeholder for Staff Creation Page for Course ${id}` until the page is ready.

**Review by pconrad — CHANGES_REQUESTED** [review]
The URL for the create course button is incorrect.  See below.

File: frontend/src/main/pages/StaffIndexPage.js (hunk 1 above)
Comment (Peak1260) [reply frontend/src/main/pages/StaffIndexPage.js]: the changes have been made accordingly

**Review by crosscountrycoder — COMMENTED** [review]
Looks good to me!

**Review by pconrad — CHANGES_REQUESTED** [review]
Why are we asking the user to enter the course id when the course id is already embedded in the URL?
image

Instead, I think we might want to remove course id and just set it from the value of course id from `useParams`

**Review by crosscountrycoder — APPROVED** [review]
Looks good to me - approved!

**Review by pconrad — APPROVED** [review]
LGTM

**Pull request merged**